.pipelines/templates/template-job-deploy-azure-env-tag.yml

```diff
@@ -0,0 +1,128 @@
+# Azure DevOps Job deploying rp
+parameters:
+  aroVersionStorageAccount: ''
+  azureDevOpsE2EJSONSPN: ''
+  azureDevOpsJSONSPN: ''
+  billingE2EPipelineName: ''
+  billingE2EBranchName: ''
+  configFileName: ''
+  e2eSubscription: ''
+  environment: ''
+  locations: []
+  rpMode: ''
+  vsoProjectID: ''
+  vsoConfigPipelineID: ''
+  vsoConfigBuildID: ''
+  tag: ''
+  rpImageAcr: ''
+
+jobs:
+- ${{ each location in  parameters.locations }}:
+  - deployment: Deploy_${{ location }}
+    variables:
+    - template: ../vars.yml
+    pool:
+      name: ARO-CI
+    environment: ${{ parameters.environment }}
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - task: DownloadBuildArtifacts@0
+            inputs:
+              buildType: specific
+              project: ${{ parameters.vsoProjectID }}
+              pipeline: ${{ parameters.vsoConfigPipelineID }}
+              ${{ if eq(parameters.vsoConfigBuildID, 'latest') }}:
+                buildVersionToDownload: latestFromBranch
+                branchName: refs/heads/master
+              ${{ if ne(parameters.vsoConfigBuildID, 'latest') }}:
+                buildVersionToDownload: specific
+                buildId: ${{ parameters.vsoConfigBuildID }}
+              downloadType: specific
+              downloadPath: $(System.ArtifactsDirectory)/config
+            displayName: Download Config
+
+          - template: ./templates/template-az-cli-login.yml
+            parameters:
+              azureDevOpsJSONSPN: $(aro-v4-ci-devops-spn)
+          - script: |
+              set -e
+              trap 'set +e; for c in $(docker ps -aq); do docker rm -f $c; done; docker image prune -af ; rm -rf ~/.docker/config.json' EXIT
+
+              export TAG=${{ parameters.vsoProjectID }}
+              export RP_IMAGE_ACR=${{ parameters.rpImageAcr }}
+
+              az acr login --name "${RP_IMAGE_ACR}"
+
+              docker pull "${RP_IMAGE_ACR}.azurecr.io/aro:${TAG}"
+              CONTAINER=$(docker create "${RP_IMAGE_ACR}.azurecr.io/aro:${TAG}")
+              docker cp $CONTAINER:/usr/local/bin/aro $(System.ArtifactsDirectory)/deployer/aro
+            displayName: Download aro deployer from image
+          - template: ./templates/template-az-cli-logout.yml
```

Comment:
Replaces pull from pipeline artifact store. Uses parameters to configure the pull